# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: editor/libeditor/tests/test_bug490879.html
# Commit: d6c4fdcf6a69
# Full Hash: d6c4fdcf6a69ebaa1b133ef8c57da9eef389f89a
# Author: Tom Schuster <evilpies@gmail.com>
# Date: 2022-10-20 21:51:26
# Regressor Bug: 1778565
# File Overlap Count: 1
# Description:
#   Bug 1778565 - Editor: Handle nsIFile transferable as blob. r=nika,masayuki
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D155753
# ==============================================================================

diff -r 88976c36da0a -r d6c4fdcf6a69 editor/libeditor/tests/test_bug490879.html
--- a/editor/libeditor/tests/test_bug490879.html	Thu Oct 20 06:59:31 2022 -0400
+++ b/editor/libeditor/tests/test_bug490879.html	Thu Oct 20 11:11:59 2022 +0000
@@ -7,7 +7,7 @@
 <iframe id="i1" width="200" height="100" src="about:blank"></iframe>
 <img id="i" src="green.png">
 <script>
-function runTest() {
+async function runTest() {
   function verifyContent() {
     const kExpectedImgSpec = "data:image/png;base64,";
     var e = document.getElementById("i1");
@@ -16,7 +16,7 @@
        kExpectedImgSpec, "The pasted image is a base64-encoded data: URI");
   }
 
-  function pasteInto() {
+  async function pasteInto() {
     var e = document.getElementById("i1");
     var doc = e.contentDocument;
     doc.designMode = "on";
@@ -25,7 +25,15 @@
     selection.removeAllRanges();
     selection.selectAllChildren(doc.body);
     selection.collapseToEnd();
+
+    let input = new Promise(resolve => {
+      doc.body.addEventListener("input", resolve, {once: true})
+    });
+
     SpecialPowers.doCommand(window, "cmd_paste");
+
+    info("Waiting for input event");
+    await input;
   }
 
   function copyToClipBoard() {
@@ -34,9 +42,10 @@
   }
 
   copyToClipBoard();
-  pasteInto();
+
+  await pasteInto();
+
   verifyContent();
-
   SimpleTest.finish();
 }
 